src/stages/scan-files.ts

```diff
@@ -1,112 +1,80 @@
 import * as modReplacements from 'module-replacements';
-import {ts as sg} from '@ast-grep/napi';
-import {readFile} from 'node:fs/promises';
-import dedent from 'dedent';
-import pc from 'picocolors';
 import * as cl from '@clack/prompts';
 import {type FileReplacement} from '../shared-types.js';
-import {suggestReplacement} from '../suggest-replacement.js';
+import {cpuUsage} from 'node:process';
+import {cpus, availableParallelism} from 'node:os';
+import {Worker} from 'node:worker_threads';
+import {fork} from 'node:child_process';
```

Comment:
you should be able to use `Worker` (it looks like it is unused here)

```ts
const w = new Worker(pathToWorkerScript);
w.on('message', handleResult);
w.on('exit', startAnotherWorker);
```

Reply:
Yes but worker was bit more work to setup, just used fork to have a poc quickly, there are also still some typings to fix etc, I'll see if I can get everything finished this evening.

Reply:
Done, see 4135793cfa8f738271600694af83a1a34c2c15e6